src/libstd/sys/unix/ext.rs

```diff
@@ -130,19 +136,28 @@ impl OsStringExt for OsString {
     }
 }
 
-// Unix-specific extensions to `OsStr`.
+/// Unix-specific extensions to `OsStr`.
 pub trait OsStrExt {
-    fn from_byte_slice(slice: &[u8]) -> &OsStr;
-    fn as_byte_slice(&self) -> &[u8];
+    fn from_bytes(slice: &[u8]) -> &OsStr;
+
+    /// Get the underlying byte view of the `OsStr` slice.
+    fn as_bytes(&self) -> &[u8];
+
+    /// Convert the `OsStr` slice into a `CString`.
+    fn to_cstring(&self) -> CString;
```

Comment:
Thinking twice on this, we currently don't provide other convenience constructors for `CString` (only a `from_slice` constructor basically). In some sense this may be better placed on a `ToCString` trait, but you still need a clear signal that this is platform-specific behavior. All-in-all we may want to leave this out for now and perhaps add it back later if necessary.